pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -727,6 +760,7 @@ func (r *ReconcileComplianceScan) phaseDoneHandler(h scanTypeHandler, instance *
 			// reset phase
 			logger.Info("Resetting scan")
 			instanceCopy := instance.DeepCopy()
+			delete(instanceCopy.Annotations, compv1alpha1.ComplianceCheckCountAnnotation)
```

Comment:
We could add a check in our e2e test utilities that asserts this annotation is deleted while the suite is waiting for the scan to cycle through it's phases. That way we're protecting this from regression if it gets accidentally removed.

Reply:
@rhmdnd You mean the user manually removing the check count annotation?

Reply:
Specifically, if line 763 is removed or refactored in a way that doesn't delete the annotation, we'd know about it because we have code in the assertions to make sure it's deleted up until the scan goes into a specific phase.